src/plugins/ecs/ecs_compares.py

```diff
@@ -0,0 +1,303 @@
+		try:
+
+
+			for account in the_config['Accounts']:
+				if account['AccountName'] == profile_name:
+					sts_client = boto3.client('sts')
+					assumedRole = sts_client.assume_role(RoleArn=role_arn, RoleSessionName="AssumedRole")
+					awsKeyId = assumedRole['Credentials']['AccessKeyId']
+					awsSecretKey = assumedRole['Credentials']['SecretAccessKey']
+					awsSessionToken = assumedRole['Credentials']['SessionToken']
+					break
+	
+			session = boto3.session.Session(aws_access_key_id=awsKeyId, aws_secret_access_key=awsSecretKey,aws_session_token=awsSessionToken)
+			client = session.client("ecs", region_name=region_name)
+
+			service_paginator = client.get_paginator('list_services')
+			service_iterator = service_paginator.paginate(cluster=cluster)
+
+		except Exception, e:
+			print("Error obtaining list of ECS services for " + cluster + " (" + str(e) + ")")
+		except KeyError, e:
+			print "Key " + e + "not found"
+
+		try:
+			for service in service_iterator:
+				# Get the service info for each batch
+				services_descriptions = client.describe_services(cluster=cluster, services=service['serviceArns'])
+				for service_desc in services_descriptions['services']:
+					image = client.describe_task_definition(taskDefinition=service_desc['taskDefinition'])
+
+					# getting ecs service version and name
+					version_output = image['taskDefinition']['containerDefinitions'][0]['image']
+					version_parsed = version_output.split("/")[-1]
+					service_dot_index = version_parsed.find(':')
+
+					service_version_prefix = version_parsed[0:service_dot_index]
+					service_version_ending = version_parsed[(service_dot_index + 1):]
+
+					# detailed ecs service
+					team_service_definition = image['taskDefinition']['family']
+
+					# version_parsed, team_service_name, region_name
+					if len(version_output) > 1:
+						c_service = {"version": service_version_ending,
+									 "servicename": service_version_prefix,
+									 "service_definition": team_service_definition,
+									 "regionname": region_name,
+									 "slackchannel": slack_channel,
+									 "environment_code_name": env_code_name}
+						service_versions.append(c_service)
+
+		except Exception, e:
+			print("Error obtaining paginated services for " + str(cluster) + " (" + str(e) + ")")
+
+
+	return service_versions
+
+	# Main comparing function
+def compare_environment(team_env, master_env, j_tags):
+
+	"""""
+	Return types
+	1 - Matches Master
+	2 - Does not match master
+	3 - branch
+	"""""
+
+	result = 0
+
+	if j_tags[0] in master_env or j_tags[1] in master_env:
+		if team_env == master_env:
+			result = 1
+		else:
+			if j_tags[0] in master_env or j_tags[1] in master_env:
+				result = 2
+			else:
+				result = 3
+
+	# print " MATCH IS: "+team_env +" == " + master_env+" ==> "+str(result)
+	logging.debug("comparing %s and %s result is %s" % (team_env, master_env, result))
+	return result
+
+def finalize_service_name(service_name, service_def, environment_code_name):
```

Comment:
A comment on the purpose of this function would be handy (I kinda agree with Kev on this generically - the WHAT isn't important in comments, it's the WHY)